### ISSUE TITLE
Split low-level OpenAI translation functionality into separate gem

### ISSUE THREAD

**Issue by westonganger** [obie/instant18n#3]
I would like to see a split of this gem into 2 distinct repos. One lower level gem to simply handle calling open ai for translations with no extra fluff. Then this gem can include all the rails view helpers and rails caching and whatever high level additions you have planned.

**Comment by davidteren** [comment]
> I would like to see a split of this gem into 2 distinct repos. One lower level gem to simply handle calling open ai for translations with no extra fluff. Then this gem can include all the rails view helpers and rails caching and whatever high level additions you have planned.

Agreed. It would be great to make this more extendable so other LLMs could be used.
